x-pack/plugins/security_solution/public/common/components/agents/agent_status/agent_response_action_status.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiBadge, EuiFlexGroup, EuiFlexItem, EuiTextColor, EuiToolTip } from '@elastic/eui';
+import type { EndpointPendingActions } from '../../../../../common/endpoint/types';
+import type { ResponseActionsApiCommandNames } from '../../../../../common/endpoint/service/response_actions/constants';
+import { RESPONSE_ACTION_API_COMMAND_TO_CONSOLE_COMMAND_MAP } from '../../../../../common/endpoint/service/response_actions/constants';
+import { ISOLATING_LABEL, RELEASING_LABEL, ISOLATED_LABEL } from './endpoint/endpoint_agent_status';
+import { useTestIdGenerator } from '../../../../management/hooks/use_test_id_generator';
+
+const TOOLTIP_CONTENT_STYLES: React.CSSProperties = Object.freeze({ width: 150 });
+
+interface AgentResponseActionsStatusProps {
+  /** The host's individual pending action list as return by the pending action summary api */
+  pendingActions: EndpointPendingActions['pending_actions'];
+  /** Is host currently isolated */
+  isIsolated: boolean;
+  'data-test-subj'?: string;
+}
+
+export const AgentResponseActionsStatus = memo<AgentResponseActionsStatusProps>(
+  ({ pendingActions, isIsolated, 'data-test-subj': dataTestSubj }) => {
+    const getTestId = useTestIdGenerator(dataTestSubj);
+
+    interface PendingActionsState {
+      actionList: Array<{ label: string; count: number }>;
+      totalPending: number;
+      wasReleasing: boolean;
+      wasIsolating: boolean;
+      hasMultipleActionTypesPending: boolean;
+      hasPendingIsolate: boolean;
+      hasPendingUnIsolate: boolean;
+    }
+
+    const {
+      totalPending,
+      actionList,
+      wasReleasing,
+      wasIsolating,
+      hasMultipleActionTypesPending,
+      hasPendingIsolate,
+      hasPendingUnIsolate,
+    } = useMemo<PendingActionsState>(() => {
+      const list: Array<{ label: string; count: number }> = [];
+      let actionTotal = 0;
+      let actionTypesCount = 0;
+
+      Object.entries(pendingActions)
+        .sort()
+        .forEach(([actionName, actionCount]) => {
+          actionTotal += actionCount;
+          actionTypesCount += 1;
```

Comment:
Q: does this equal to pendingActions.length?

Reply:
Not sure I understand, `pendingActions` is an object which is why we list it out as array of it's keys and values to do these calculations

Reply:
But if I see it correctly, you do increment by 1 on every iteration. Meaning, we end up with a number equal to object values length. Is this about right?

Reply:
Sure I'll try and make this a bit more readable.